lib/src/internal/http/http_handler.dart

```diff
@@ -1,84 +1,108 @@
+import 'dart:convert';
+
 import 'package:http/http.dart' as http;
 
 import 'package:logging/logging.dart';
 import 'package:nyxx/src/events/http_events.dart';
+import 'package:nyxx/src/events/ratelimit_event.dart';
 import 'package:nyxx/src/internal/event_controller.dart';
+import 'package:nyxx/src/internal/exceptions/http_client_exception.dart';
 import 'package:nyxx/src/nyxx.dart';
 import 'package:nyxx/src/internal/http/http_bucket.dart';
 import 'package:nyxx/src/internal/http/http_request.dart';
 import 'package:nyxx/src/internal/http/http_response.dart';
+import 'package:nyxx/src/utils/utils.dart';
 
 class HttpHandler {
-  final RegExp _bucketRegexp = RegExp(r"\/(channels|guilds)\/(\d+)");
-  final RegExp _bucketReactionsRegexp = RegExp(r"\/channels/(\d+)\/messages\/(\d+)\/reactions");
-  final RegExp _bucketCommandPermissions = RegExp(r"\/applications/(\d+)\/guilds\/(\d+)\/commands/permissions");
-
-  final List<HttpBucket> _buckets = [];
-  late final HttpBucket _noRateBucket;
-
   late final http.Client httpClient;
 
   final Logger logger = Logger("Http");
   final INyxxRest client;
 
+  RestEventController get _events => client.eventsRest as RestEventController;
+
+  final Map<String, HttpBucket?> _bucketByRequestRateLimitId = <String, HttpBucket?>{};
```

Comment:
No need for the type parameters on the map literal here, type inference does it for us